universal-application-tool-0.0.1/app/services/program/ProgramDefinition.java

```diff
@@ -179,8 +175,7 @@ public boolean hasQuestion(long questionId) {
               blockDefinitions().stream()
                   .map(BlockDefinition::programQuestionDefinitions)
                   .flatMap(ImmutableList::stream)
-                  .map(ProgramQuestionDefinition::getQuestionDefinition)
-                  .map(QuestionDefinition::getId)
+                  .map(ProgramQuestionDefinition::id)
```

Comment:
Question (blocking): are `ProgramQuestionDefinition`'s guaranteed to have the same ID as the `QuestionDefinition`'s they contain?

Reply:
I think so, ProgramQuestionDefinition basically is a wrapper for serializing QuestionDefinition id as json.

Reply:
Yep! The code I'm replacing here assumes you got this object from the `Service` which fills in those definitions.